balloon/src/main/kotlin/com/skydoves/balloon/Balloon.kt

```diff
@@ -1602,6 +1602,15 @@ public class Balloon private constructor(
             onBalloonOutsideTouchListener?.onBalloonOutsideTouch(view, event)
             return true
           }
+          if (event.action == MotionEvent.ACTION_UP) {
+            if (binding.balloonWrapper.getViewPointOnScreen().x > event.rawX) {
+              if (builder.dismissWhenTouchOutside) {
+                this@Balloon.dismiss()
+              }
+              onBalloonOutsideTouchListener?.onBalloonOutsideTouch(view, event)
+              return true
+            }
+          }
```

Comment:
This is an excellent idea 👏 I'd never thought about this way to address this issue.

What about considering touching the right side of the balloon?

Maybe this one should work as the same approach:

```kotlin
if (binding.balloonWrapper.getViewPointOnScreen().x > event.rawX ||
   binding.balloonWrapper.getViewPointOnScreen().x + getMeasuredWidth() < event.rawX
) {
  ..
}
```

Potentially, this won't resolve the top-bottom margin cases, but we can address both the left and right cases.

Reply:
Thank you @skydoves for generous feedback. I am glad you liked the approach taken in this PR.

I am not sure whether to add this code inside   **this.bodyWindow.setTouchInterceptor()** or have a separate listener and a field like **dismissWhenTouchMargin** . This way people that care about dismissing on margin touch can utilise the functionality, while others can safely ignore it.

Reply:
Having a separate `dismissWhenTouchMargin` field sounds great. We can take this value as `true` by default.

Reply:
Sorry for the late reply @skydoves.
Hectic couple of days at work. I managed to get to setDismissWhenTouchMargin only now, was not sure how to code it properly. However it seems you have already implemented it. Thank you for making my life easier :D

Reply:
No problem! Everything looks very good to me. Again, thank you so much for your contribution, and happy holidays!

Reply:
Thank you and happy holidays to you too